hazelcast/src/main/java/com/hazelcast/instance/DefaultHazelcastInstanceFactory.java

```diff
@@ -19,10 +19,12 @@
 import com.hazelcast.config.Config;
 import com.hazelcast.config.XmlConfigBuilder;
 import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.spi.annotation.PrivateApi;
 
 /***
  * Default implementation for Hazelcast instance factory;
  */
+@PrivateApi
```

Comment:
If the interface is meant to be implemented by external services there may also be implementations which extend this non-final class?